# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/prim_shared.glsl
# Commit: 15da94c6a7b1
# Full Hash: 15da94c6a7b19f971d3dffaa4cab8edb8e41cb62
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-15 09:29:54
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r 72ddac486037 -r 15da94c6a7b1 gfx/wr/webrender/res/prim_shared.glsl
--- a/gfx/wr/webrender/res/prim_shared.glsl	Tue Dec 15 01:46:51 2020 +0200
+++ b/gfx/wr/webrender/res/prim_shared.glsl	Mon Dec 14 21:48:41 2020 +0000
@@ -15,7 +15,7 @@
 #define RASTER_LOCAL            0
 #define RASTER_SCREEN           1
 
-uniform sampler2DArray sClipMask;
+uniform sampler2D sClipMask;
 
 vec2 clamp_rect(vec2 pt, RectWithSize rect) {
     return clamp(pt, rect.p0, rect.p0 + rect.size);
@@ -266,11 +266,7 @@
         return 0.0;
     }
     // finally, the slow path - fetch the mask value from an image
-    // Note the Z getting rounded to the nearest integer because the variable
-    // is still interpolated and becomes a subject of precision-caused
-    // fluctuations, see https://bugzilla.mozilla.org/show_bug.cgi?id=1491911
-    ivec3 tc = ivec3(mask_uv, vClipMaskUv.z + 0.5);
-    return texelFetch(sClipMask, tc, 0).r;
+    return texelFetch(sClipMask, ivec2(mask_uv), 0).r;
 }
 
 #ifdef WR_FEATURE_DITHERING